internal/cache/metadata/stat_cache.go

```diff
@@ -247,8 +247,19 @@ func (sc *statCacheBucketView) LookUpFolder(
 	now time.Time) (bool, *gcs.Folder) {
 	// Look up in the LRU cache.
 	hit, entry := sc.sharedCacheLookup(folderName, now)
+
 	if hit {
-		return hit, entry.f
+		// Adds negative entry scenario
+		if entry.f == nil && entry.m == nil {
```

Comment:
Are there older tests for stat cache negative scenario and are they passing?

Reply:
yes *StatCacheTest* has multiple stat cache negative scenarios for both object and folders which is passing for this PR.